pkg/kubelet/cm/cpumanager/state/state_checkpoint_test.go

```diff
@@ -423,6 +423,88 @@ func TestCheckpointStateClear(t *testing.T) {
 	}
 }
 
+func TestCheckpointStateAllocateAndReclaim(t *testing.T) {
+	testCases := []struct {
+		description   string
+		defaultCPUset cpuset.CPUSet
+		assignments   map[string]map[string]cpuset.CPUSet
+	}{
+		{
+			description:   "One container",
+			defaultCPUset: cpuset.New(0, 1, 2, 3, 4, 5, 6, 7, 8),
+			assignments: map[string]map[string]cpuset.CPUSet{
+				"pod": {
+					"c1": cpuset.New(0, 1),
+				},
+			},
+		},
+	}
+
+	// create temp dir
+	testingDir, err := os.MkdirTemp("", "cpumanager_state_test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer func() {
+		if rmErr := os.RemoveAll(testingDir); rmErr != nil {
+			t.Fatalf("could not remove checkpoint: %v", rmErr)
+		}
+	}()
+
+	cpm, err := checkpointmanager.NewCheckpointManager(testingDir)
+	if err != nil {
+		t.Fatalf("could not create testing checkpoint manager: %v", err)
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.description, func(t *testing.T) {
+			// ensure there is no previous checkpoint
+			if rmErr := cpm.RemoveCheckpoint(testingCheckpoint); rmErr != nil {
+				t.Fatalf("could not remove previous checkpoint: %v", rmErr)
+			}
+
+			logger, _ := ktesting.NewTestContext(t)
+			state, err := NewCheckpointState(logger, testingDir, testingCheckpoint, "none", nil)
+			if err != nil {
+				t.Fatalf("could not create testing checkpointState instance: %v", err)
+			}
+
+			defaultCPUset := tc.defaultCPUset
+			assignments := tc.assignments
+			state.SetDefaultCPUSet(defaultCPUset)
+
+			for pod := range assignments {
+				for container, set := range assignments[pod] {
+					state.Allocate(pod, container, set)
+
+					var cpus cpuset.CPUSet
+					if cpus, _ = state.GetCPUSet(pod, container); !cpus.Equals(set) {
+						t.Fatalf("state inconsistent, got %q instead of %q", set, cpus)
+					}
```

Comment:
Please, consider using assert/require here and below, e.g.

```suggestion
					cpus, _ = state.GetCPUSet(pod, container)
					require.Equal(t, set, cpus)
```